components/server/src/user/user-service.ts

```diff
@@ -248,6 +249,36 @@ export class UserService {
         return AttributionId.render({ kind: "team", teamId: project.teamId });
     }
 
+    async setUsageAttribution(user: User, usageAttributionId: string | undefined): Promise<void> {
+        if (typeof usageAttributionId !== "string") {
+            user.usageAttributionId = undefined;
+            await this.userDb.storeUser(user);
+            return;
+        }
+        const attributionId = AttributionId.parse(usageAttributionId);
+        if (attributionId?.kind === "user") {
+            if (user.id === attributionId.userId) {
+                user.usageAttributionId = usageAttributionId;
+                await this.userDb.storeUser(user);
+                return;
+            }
+            throw new Error("Cannot attribute another user.");
```

Comment:
Please classify the error as Permission Denied

Reply:
Agree.